### PR TITLE
Add loader at appropriate places #14

### LINKED ISSUE #14: Add loader at appropriate places
We need material spinners at different places around the site. For eg: when a component is being loaded or a service is being waited for.

Add the following code to add spinner:
`<div class="loader" *ngIf="!isPageLoaded"><mat-spinner></mat-spinner></div>`
and declare the variable `isPageLoaded=false` in the respective `component.ts` file. This variable will be changed to `true` once the response is complete.

Look at the file [details.component.ts](https://github.com/Cynthesize/frontend/blob/master/src/app/profile/user/details/details.component.ts) and [details.component.html](https://github.com/Cynthesize/frontend/blob/master/src/app/profile/user/details/details.component.html) for an example.

### PR DESCRIPTION
**Pull request by neozenweb**
Fixes #14  
Added spinner loader for 4 components(changes to html and ts):
src/app/add/add-idea components
src/app/add/project components
src/app/shared/comments components
src/app/shared/liking components

### CODE REVIEW

**Comment by neozenweb** [comment]
Just saw the wiki and it suggests the following steps
git checkout master
	git fetch upstream
	git merge upstream/master
	git push origin master

Since I do not have merge permissions, I created the pull request without these steps. Should I start another PR after these steps. Thanks.

**Comment by WickedBrat** [comment]
@neozenweb Here are a few suggestions about the PR
- In order to fetch and merge upstream, you don't need merge permissions. With the command `git merge upstream/master` you're merging your origin with the master branch of the upstream. It should work fine. When you run the command, it'll show you merge conflicts that you'll need to resolve manually and commit them after.
- Choose the title of the PR to something relevant, like `Adds preloader for API calls`
- The PR description must start from "Fixes <issue-number>" and later you should mention the changes you made in the PR

**Comment by neozenweb** [comment]
@WickedBrat  Thanks for your patience.  Here is the error message I get on using
 git fetch upstream
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.

**Comment by WickedBrat** [comment]
@neozenweb Probably the errors you're facing are the ones that have remained unchanged from the commits on the repo. You'd always want the latest changes that are in the repo in your repo as well. Such that the changes that are reflected in the PRs are only the ones that are relevant to your PR. While fixing the conflicts, choose the changes should be reflected in the PR. For example: Refer the images below, I've fixed the commits for `idea.component.ts`
screenshot from 2019-02-15 03-48-21
screenshot from 2019-02-15 03-48-36

**Comment by neozenweb** [comment]
Went through each file difference and fixed them all for commit. But again the deploy failed. Do you want me to revert the pull and get the latest code and redo the changes?

File: package.json [hunk 1]
```diff
@@ -91,7 +91,7 @@
     "karma-jasmine": "^2.0.1",
     "karma-junit-reporter": "^1.2.0",
     "prettier": "^1.16.4",
-    "pretty-quick": "^1.10.0",
+    "pretty-quick": "^1.6.0",
```
Comment (WickedBrat) [inline package.json]: Update this.

File: src/app/shared/comments/comments.component.ts [hunk 2]
```diff
@@ -98,18 +103,24 @@ export class CommentsComponent implements OnInit, OnChanges {
...
   cancel(): void {
     this.commentingOnIssue = false;
+    this.isLoading = false;
   }
 
   fetchIdeaComments() {
+    this.isLoading = true;
     this.ideaService.getIdeaComments(this.activityId).subscribe(
       (data: any) => {
+        console.log(data);
```
Comment (WickedBrat) [inline src/app/shared/comments/comments.component.ts]: remove this line

File: package.json (hunk 1 above)
Comment (neozenweb) [reply package.json]: updated.

File: src/app/shared/comments/comments.component.ts (hunk 2 above)
Comment (neozenweb) [reply src/app/shared/comments/comments.component.ts]: Done.

**Comment by neozenweb** [comment]
Done with the changes. Thanks for your help @WickedBrat . Should the files like package.json and package-lock.json be in my .gitignore files, so that they dont upload everytime?

**Comment by WickedBrat** [comment]
@neozenweb Yes, it's better to add package-lock.json in .gitignore. You may do this in a separate PR.

**Comment by neozenweb** [comment]
sorry siddhant, I fixed the code and clicked on "resolve conversation" ...didn't realize that would close the PR

**Comment by WickedBrat** [comment]
@neozenweb I'm not sure what exactly is the problem here. I'd suggest you creating a new PR anyway. Let's see if that fixes it

**Comment by neozenweb** [comment]
Sure.  I will create a new PR.

**Pull request closed without merging**